language/stdlib/modules/DesignatedDealer.move

```diff
@@ -100,10 +101,12 @@ module DesignatedDealer {
             window_inflow: 0,
             tiers: Vector::empty(),
         });
-        add_tier<CoinType>(tc_account, dd_addr, TIER_0_DEFAULT);
-        add_tier<CoinType>(tc_account, dd_addr, TIER_1_DEFAULT);
-        add_tier<CoinType>(tc_account, dd_addr, TIER_2_DEFAULT);
-        add_tier<CoinType>(tc_account, dd_addr, TIER_3_DEFAULT);
+        // Add tier amounts in base_units of CoinType
+        let coin_scaling_factor = Libra::scaling_factor<CoinType>();
+        add_tier<CoinType>(tc_account, dd_addr, TIER_0_DEFAULT * coin_scaling_factor);
```

Comment:
This is safe because the scaling factor is immutable, right?

Reply:
Yep! Good callout, scaling factor is an immutable field in currency info.

Reply:
👍 . There's no way to make data in move always immutable is there?

Reply:
In the current version of the stdlib there is no way to mutate this field, so we have immutability via the module's public functions. But, the module could be updated to mutate it, or a writeset could happen. However both of these are incredibly unlikely, and we would have to factor this if mutation is performed/allowed at a latter date.